exercises/practice/bank-account/bank-account-test.el

```diff
@@ -0,0 +1,152 @@
+;;; bank-account-test.el --- Bank Account (exercism)  -*- lexical-binding: t; -*-
+
+;;; Commentary:
+
+;;; Code:
+
+
+(load-file "bank-account.el")
+
+(require 'eieio)
+
+(ert-deftest newly-opened-account-has-zero-balance ()
+  (let ((account (make-new-bank-account)))
+    (open-account account)
+    (should (= (balance account) 0))))
+
+
+(ert-deftest single-deposit ()
+  (let ((account (make-new-bank-account)))
+    (open-account account)
+    (deposit account 100)
+    (should (= (balance account) 100))))
+
+
+(ert-deftest multiple-deposits ()
+  (let ((account (make-new-bank-account)))
+    (open-account account)
+    (deposit account 100)
+    (deposit account 50)
+    (should (= (balance account) 150))))
+
+
+(ert-deftest withdraw-once ()
+  (let ((account (make-new-bank-account)))
+    (open-account account)
+    (deposit account 100)
+    (withdraw account 75)
+    (should (= (balance account) 25))))
+
+
+(ert-deftest withdraw-twice ()
+  (let ((account (make-new-bank-account)))
+    (open-account account)
+    (deposit account 100)
+    (withdraw account 80)
+    (withdraw account 20)
+    (should (= (balance account) 0))))
+
+
+(ert-deftest can-do-multiple-operations-sequentially ()
+  (let ((account (make-new-bank-account)))
+    (open-account account)
+    (deposit account 100)
+    (deposit account 110)
+    (withdraw account 200)
+    (deposit account 60)
+    (withdraw account 50)
+    (should (= (balance account) 20))))
+
+
+(ert-deftest cannot-check-balance-of-closed-account ()
```

Comment:
Similar to the circular-buffer PR, we might want to have students create custom error types and test for those here instead of checking the contents directly.

Reply:
Agreed. I will look into it.

Reply:
done. I think I will have to rebase again if we merge circular-buffers first.